mlflow/R/mlflow/R/python.R

```diff
@@ -23,13 +27,26 @@ python_bin <- function() {
 # Returns path to MLflow CLI, assumed to be in the same bin/ directory as the
 # Python executable
 python_mlflow_bin <- function() {
+  in_env <- Sys.getenv("MLFLOW_BIN")
+  if (file.exists(in_env)) {
+    return(in_env)
+  }
   python_bin_dir <- dirname(python_bin())
-  file.path(python_bin_dir, "mlflow")
+  path <- file.path(python_bin_dir, "mlflow")
+  if (!file.exists(path) && file.exists(Sys.getenv("MLFLOW_PYTHON_BIN"))) {
+    stop("Mlflow executable does not seem to be in the same directory than Python.\n",
+         "  Please set the environment variable MLFLOW_BIN to the path of your mlflow executable."
+         )
+  }
+  path
 }
 
 # Return path to conda home directory, such that the `conda` executable can be found
 # under conda_home/bin/
 #' @importFrom reticulate conda_binary
 python_conda_home <- function() {
+  if (file.exists(Sys.getenv("MLFLOW_PYTHON_BIN")) && file.exists(python_mlflow_bin())) {
```

Comment:
Can we call mlflow_conda_bin() in Try/Catch block and return NULL on exception? The existence of conda does not depend on if user selected other python binary.